src/app/test_executive/bootstrap_test.ml

```diff
@@ -0,0 +1,40 @@
+open Async
+open Core
+open Integration_test_lib
+
+module Make (Engine : Engine_intf) = struct
+  open Engine
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type log_engine = Log_engine.t
+
+  let config =
+    let open Test_config in
+    let open Test_config.Block_producer in
+    { default with
+      block_producers= [{balance= "1000"}; {balance= "1000"}]
+    ; num_snark_workers= 0 }
+
+  let run network log_engine =
+    let open Network in
+    let open Deferred.Or_error.Let_syntax in
+    let node_a = List.nth_exn network.block_producers 0 in
+    let node_b = List.nth_exn network.block_producers 1 in
+    (* TODO: bulk wait for init *)
+    let%bind () = Log_engine.wait_for_init node_a log_engine in
+    let%bind () = Log_engine.wait_for_init node_b log_engine in
+    let%bind () =
+      Log_engine.wait_for ~blocks:2 ~timeout:(`Slots 20) log_engine
```

Comment:
Should we wait till `should_bootstrap_len`(in `proof_of_stake.ml`) blocks?

Reply:
For a more robust bootstrap test, yes. I was tasking out multiple different scopes for bootstrap tests: a short one which just tests the basic functionality of bootstrapping from genesis to a short chain, a longer one which exercises bootstrapping from genesis to a longer chain (which would need to be `> should_bootstrap_len` to be useful), and a longer one which exercises bootstrapping from various persisted states. Do you think it makes sense to put these in different tests we run, or to have one test which checks all of these, and just spend the time trying to make that test run fast (short slots and epochs, proof level check, etc)?

Reply:
Yeah, it makes sense to have separate tests for each of the cases. It is faster that way and easier to debug in case of failures.